.github/workflows/ci.yaml

```diff
@@ -95,15 +98,47 @@ jobs:
           path: /tmp/.buildx-cache
           key: ${{ runner.os }}-buildx-${{ github.sha }}
           restore-keys: |
-            ${{ runner.os }}-buildx-cache-ci
+            ${{ runner.os }}-buildx-${{ steps.extract_branch.outputs.branch }}
 
-      #todo : move this step from the makefile to the CI makefile.
       - name: Fix permissions
-        run: make fix-permissions
+        run: make  -C .github/scripts fix-permissions
+
+      - id: cache-docker
+        uses: actions/cache@v1
+        with:
+          path: /tmp/docker-registry
+          key: docker-registry-${{ steps.extract_branch.outputs.branch }}
+
+      - run: docker run -d -p 5000:5000 --restart=always --name registry -v /tmp/docker-registry:/var/lib/registry registry:2
```

Comment:
Why not use a GitHub actions service for running this? Or simply do `docker image save`, which would allow you to download and inspect the images as well.

Reply:
I did not found a fitting action. Docker image save appears to be slower, but indeed it would allow to inspect the images..

Reply:
@ctron how about we export the images to an artefact if there is a failure